rinja_parser/src/lib.rs

```diff
@@ -274,18 +274,51 @@ impl<'a> From<ErrorContext<'a>> for winnow::error::ErrMode<ErrorContext<'a>> {
     }
 }
 
-fn is_ws(c: char) -> bool {
-    matches!(c, ' ' | '\t' | '\r' | '\n')
+/// FIXME: replace with `s.trim_ascii_start()` when we raise our MSRV to 1.80
+fn trim_ascii_start(s: &str) -> &str {
+    let mut s = s.as_bytes();
+    loop {
+        match s {
+            [c, tail @ ..] if c.is_ascii_whitespace() => s = tail,
+            // SAFETY: either we are at the front of the input, at an empty input, or the previous
+            // character was an ASCII character, so we must be at a char boundary
+            s => return unsafe { std::str::from_utf8_unchecked(s) },
+        }
+    }
+}
```

Comment:
The same code is generated as with the built-in method: [godbolt]

[godbolt]: <https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:rust,selection:(endColumn:1,endLineNumber:16,positionColumn:1,positionLineNumber:16,selectionStartColumn:1,selectionStartLineNumber:16,startColumn:1,startLineNumber:16),source:'%23%5Binline(never)%5D%0Apub+fn+trim_ascii_start1(s:+%26str)+-%3E+%26str+%7B%0A++++let+mut+s+%3D+s.as_bytes()%3B%0A++++loop+%7B%0A++++++++match+s+%7B%0A++++++++++++%5Bc,+tail+@+..%5D+if+c.is_ascii_whitespace()+%3D%3E+s+%3D+tail,%0A++++++++++++s+%3D%3E+return+unsafe+%7B+std::str::from_utf8_unchecked(s)+%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A%0A%23%5Binline(never)%5D%0Apub+fn+trim_ascii_start2(s:+%26str)+-%3E+%26str+%7B%0A++++s.trim_ascii_start()%0A%7D%0A'),l:'5',n:'0',o:'Rust+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:r1820,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'1',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:rust,libs:!(),options:'-C++++++++++++++++opt-level%3D3',overrides:!(),selection:(endColumn:27,endLineNumber:18,positionColumn:27,positionLineNumber:18,selectionStartColumn:27,selectionStartLineNumber:18,startColumn:27,startLineNumber:18),source:1),l:'5',n:'0',o:'+rustc+1.82.0+(Editor+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4>